enhancements/ingress/haproxy-router-compression.md

```diff
@@ -173,38 +125,46 @@ type IngressControllerSpec struct {
     HTTPCompression *HTTPCompressionPolicy `json:"httpCompression,omitempty"`
 }
 ...
-// httpCompressionPolicy defines the compression algorithm and MIME types that need to
-// have compression applied.
+// httpCompressionPolicy turns on compression for the specified MIME types
 //
 // This field is optional, and its absence implies that compression should not be enabled
 // globally in HAProxy.
 //
-// If httpCompressionPolicy exists, compression should be enabled only for the listed
-// MIME types, and only using the designated compression algorithm.
+// If httpCompressionPolicy exists, compression should be enabled only for the specified
+// MIME types
 type HTTPCompressionPolicy struct {
 
-	// compressionAlgorithm defines the desired compression algorithm.
+	// compressionEnabled defines the desired compression enablement
+	// If true, compression is enabled for all MIME types specified in
+	// CompressionMIMETypes.  If false, compression is disabled for any MIME type.
 	//
 	// +required
-	CompressionAlgorithm CompressionAlgorithmType `json:"compressionAlgorithm,omitempty"`
+	// +kubebuilder:default=false
+	CompressionEnabled bool `json:"compressionEnabled"`
```

Comment:
Do we need this if we have entries in `CompressionMIMETypes`. If `CompressionMIMETypes` is empty then compression is disabled. I don't know whether users would want the convenience of toggling `CompressionEnabled` vis-a-vis just removing the list of mime types. Thoughts?

Reply:
I agree, we don't need this. Removed.  @frobware @miciah PTAL at latest change (minor) and tag if lgty.